src/System.IO.FileSystem/tests/File/EncryptDecrypt.cs

```diff
@@ -16,11 +18,113 @@ public static void NullArg_ThrowsException()
+            File.WriteAllText(tmpFileName, textContentToEncrypt);
+            try
+            {
+                //check if runas works
+                Whoami();
+
+                File.Encrypt(tmpFileName);
+                Assert.Equal(textContentToEncrypt, File.ReadAllText(tmpFileName));
+
+                //try to read encrypted file as other user
+                (int exitCode, string contentRead) readResult = ReadAsOtherUser(tmpFileName);
+                Assert.NotEqual(0, readResult.exitCode);
+                Assert.Empty(readResult.contentRead);
+
+                File.Decrypt(tmpFileName);
+                Assert.Equal(textContentToEncrypt, File.ReadAllText(tmpFileName));
+
+                //try to read unecrypted file as other user
+                readResult = ReadAsOtherUser(tmpFileName);
+                Assert.Equal(0, readResult.exitCode);
+                Assert.Equal(textContentToEncrypt, readResult.contentRead);
+            }
+            finally
+            {
+                File.Delete(tmpFileName);
+            }
+
+            void Whoami()
+            {
+                ProcessStartInfo psiWhoami = new ProcessStartInfo();
+                psiWhoami.Verb = "runas";
+                psiWhoami.LoadUserProfile = true;
+                psiWhoami.FileName = "cmd.exe";
+                psiWhoami.Arguments = $"/C whoami";
+                psiWhoami.UseShellExecute = false;
+                psiWhoami.UserName = EncryptDecryptFixture.UserName;
+                psiWhoami.Password = EncryptDecryptFixture.SecureStringPassword;
+                psiWhoami.RedirectStandardOutput = true;
+                Process pWhoami = Process.Start(psiWhoami);
+                pWhoami.WaitForExit();
+                Assert.Equal((Environment.MachineName + "\\" + EncryptDecryptFixture.UserName).ToUpper(), pWhoami.StandardOutput.ReadToEnd().ToUpper().Trim());
+            }
+
+            (int exitCode, string contentRead) ReadAsOtherUser(string filePath)
+            {
+                ProcessStartInfo piReadFileContent = new ProcessStartInfo();
+                piReadFileContent.Verb = "runas";
+                piReadFileContent.LoadUserProfile = true;
+                piReadFileContent.FileName = "cmd.exe";
+                piReadFileContent.Arguments = $"/C type \"{filePath}\"";
+                piReadFileContent.UseShellExecute = false;
+                piReadFileContent.UserName = EncryptDecryptFixture.UserName;
+                piReadFileContent.Password = EncryptDecryptFixture.SecureStringPassword;
+                piReadFileContent.RedirectStandardOutput = true;
+                Process processReadFileContent = Process.Start(piReadFileContent);
+                processReadFileContent.WaitForExit();
+                return (processReadFileContent.ExitCode, processReadFileContent.StandardOutput.ReadToEnd());
+            }
+        }
+    }
+
+    public class EncryptDecryptFixture : IDisposable
```

Comment:
Can you separate the different user stuff into a separate PR? I don't want to hold this up on what it likely to be a more lengthy review. A few initial comments:

1. Anything we add that spawns a separate process should ideally be added to the RemoteExecutorTestBase
2. We'd have to thoroughly vet adding a new user account (or accounts) to the machine- there are a bunch of questions here (security/cleanup/collision/etc.)
3. There might be a simpler way to do this- I think we might be able to use CreateRestrictedToken (via P/Invoke) and WindowsIdentity.RunImpersonated to avoid even going out of process. I'm looking into it and will update. I'm hoping that copying the process token and disabling the current user SID will be enough to effectively validate.

Reply:
Ok!